### PR TITLE
New functionality to show or hide slices of pie chart

### PR DESCRIPTION
**Pull request by clancelotti**
issue #12315 
Signed-off-by: Cristiano Lancelotti <lancelot.info@gmail.com>

Before submitting pull request, please check that every commit:

- [x] Has proper Signed-Off-By
- [x] Has commit message which describes it
- [x] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [x] Any new functionality is covered by tests

Summary of changes:
- Add plugin EnhancedPieLegendRenderer that allow extending the functionalities of the legend: clicking the legend items, you can show or hide the corresponding slices of pie chart.
- Add dependency on files
- Activate the plug-in in options 
- Adjust the position of legend placed on server status query pie chart.

The changes allow the server status query pie chart and the SQL query pie chart show or hide each slice. 
Current behaviour: when clicking on each legend item, the page remains loading and nothing happens.
After changes: when clicking on each legend item, the corresponding chart slice toggles between showing and hiding.

I have tested it on Internet Explorer, Mozilla Firefox and Google Chrome. It works fine here. Take a look, please.

### CODE REVIEW

**Comment by clancelotti** [comment]
Applying the above changes (commit 5781a00), we have a gray shadow pie chart (defined configuration before these proposed changes). I think it's not cool because it looks like a gray slice of the chart when hiding a slice of another color. 
The next images are with shadow yet and I hid some elements to see the problem:

server_status_pie_chart


sql_pie_chart

Then I set to false the value of shadow and it is not displayed anymore.
Another problem occurs just with status queries pie chart. There are a defined list of colors used to apply on it. When there is a exhibited white slice (without shadow), it's difficult to make difference between the slice color and the background color of the chart. Therefore, I deleted color #eeeeec. Besides that, I added a slice margin, 
See the images with those changes (commit 37d39179):

server_status_pie_chart_change

sql_pie_chart_change

File: themes/pmahomme/css/common.css.php [hunk 1]
```diff
@@ -1296,7 +1296,7 @@
     float: <?php echo $left; ?>;
     width: 500px;
     height: 350px;
-    padding-<?php echo $left; ?>: 30px;
+    margin-right: 50px;
```
Comment (nijel) [inline themes/pmahomme/css/common.css.php]: This will break in RTL languages.

File: themes/original/css/common.css.php [hunk 2]
```diff
@@ -974,10 +974,10 @@
...
 
 div#serverstatusquerieschart {
-    float:<?php echo $left; ?>;
-    width:500px;
-    height:350px;
-    padding-<?php echo $left; ?>: 30px;
+    float: <?php echo $left; ?>;
+    width: 500px;
+    height: 350px;
+    margin-right: 50px;
```
Comment (nijel) [inline themes/original/css/common.css.php]: This will break in RTL languages.

File: js/functions.js [hunk 3]
```diff
@@ -2516,7 +2516,6 @@ function PMA_createProfilingChart(target, data)
             '#729fcf',
             '#ad7fa8',
             '#ef2929',
-            '#eeeeec',
```
Comment (nijel) [inline js/functions.js]: Why removing one color here?

Comment (clancelotti) [inline js/functions.js]: Thanks for the feedback. This color is very similar to the background color. See the image (other element). What do you think? 
removed color

File: themes/pmahomme/css/common.css.php (hunk 1 above)
Comment (clancelotti) [inline themes/pmahomme/css/common.css.php]: In this case, a minor change in 1299: padding-<?php echo $left; ?>: 30px; is enough to adjust the legend position. I tested changing 30px to 5 px and it's OK. Any suggestion of other solution that does not break RTL?

Comment (nijel) [inline themes/pmahomme/css/common.css.php]: Use `margin-<?php echo $right;?>` instead, so that it's properly oriented in RTL languages.

**Comment by nijel** [comment]
Thanks for fixing those. Still I'm not sure this is something what would help in the original report:

localhost_localhost_phpmyadmin_4 8 0-dev_-_2017-02-14_13 22 35

I'd rather expect the removed parts not showing up at all in the chart - the remaining pieces should IMHO expand to cover full pie instead of just part of it...

**Comment by clancelotti** [comment]
I have thought about it and makes sense what you said. 
I'm working on it, thanks.

**Comment by clancelotti** [comment]
@nijel, I think that it's working like you suggested:
filling all chart

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**